src/MiniProfiler.EF6/MiniProfilerEF6.cs

```diff
@@ -10,15 +13,23 @@ namespace StackExchange.Profiling.EntityFramework6
     /// </summary>
     public static class MiniProfilerEF6
     {
+        private static readonly ConcurrentDictionary<object, object>  _resolvedDependenciesCache = new ConcurrentDictionary<object, object>();
```

Comment:
Are we sure this doesn't grow unbounded?
I'd argue for 4 properly typed dictionaries here, assuming there's no unbounded growth. This also has a side-effect of caching the services even if they were to be replaced downstream later, is that a concern? e.g. the thing we're wrapping could be swapped, and we wouldn't care, as the `inner` isn't a key function.